inference/models/florence2/florence2.py

```diff
@@ -59,3 +89,33 @@ def prepare_generation_params(
             "input_ids": preprocessed_inputs["input_ids"],
             "pixel_values": preprocessed_inputs["pixel_values"],
         }
+
+    def predict(self, image_in: Image.Image, prompt="", history=None, **kwargs):
+        model_inputs = self.processor(
+            text=prompt, images=image_in, return_tensors="pt"
+        ).to(self.model.device)
+        input_len = model_inputs["input_ids"].shape[-1]
+
+        with torch.inference_mode():
+            prepared_inputs = self.prepare_generation_params(
+                preprocessed_inputs=model_inputs
+            )
+            generation = self.model.generate(
+                **prepared_inputs,
+                max_new_tokens=1024,
+                do_sample=False,
+                early_stopping=False,
+                num_beams=3
+            )
+            generation = generation[0]
+            if self.generation_includes_input:
+                generation = generation[input_len:]
+            decoded = self.processor.decode(generation, skip_special_tokens=False)
+            parsed_answer = self.processor.post_process_generation(
```

Comment:
Predict shouldn't be rewritten completely if the point of this part of PR is just to add this parsed_answer. Also, duplicate code. Use a mixin that alters the postprocess method and have both of these classes inherit from this.